moabb/paradigms/cvep.py

```diff
@@ -0,0 +1,189 @@
+import logging
+
+from moabb.datasets import utils
+from moabb.datasets.fake import FakeDataset
+from moabb.paradigms.base import BaseParadigm
+
+
+log = logging.getLogger(__name__)
+
+
+class BaseCVEP(BaseParadigm):
+    """Base c-VEP paradigm.
+
+    Please use one of the child classes.
+
+    Parameters
+    ----------
+
+    filters: list of list (defaults [[1, 40]])
+        Bank of bandpass filter to apply.
+
+    events: List of str | None (default None)
+        Event to use for epoching. If None, default to all events defined in
+        the dataset.
+
+    tmin: float (default 0.0)
+        Start time (in second) of the epoch, relative to the dataset specific
+        task interval e.g. tmin = 1 would mean the epoch will start 1 second
+        after the beginning of the task as defined by the dataset.
+
+    tmax: float | None (default None)
+        End time (in second) of the epoch, relative to the beginning of the
+        dataset specific task interval. tmax = 5 would mean the epoch will end
+        5 second after the beginning of the task as defined in the dataset. If
+        None, use the dataset value.
+
+    baseline: None | tuple of length 2 (default None)
+        The time interval to consider as “baseline” when applying baseline
+        correction. If None, do not apply baseline correction.
+        If a tuple (a, b), the interval is between a and b (in seconds),
+        including the endpoints.
+        Correction is applied by computing the mean of the baseline period
+        and subtracting it from the data (see mne.Epochs)
+
+    channels: list of str | None (default None)
+        list of channel to select. If None, use all EEG channels available in
+        the dataset.
+
+    resample: float | None (default None)
+        If not None, resample the eeg data with the sampling rate provided.
+    """
+
+    def __init__(
+        self,
+        filters=([1, 40],),
+        events=None,
+        tmin=0.0,
+        tmax=None,
+        baseline=None,
+        channels=None,
+        resample=None,
+    ):
+        super().__init__(
+            filters=filters,
+            events=events,
+            channels=channels,
+            baseline=baseline,
+            resample=resample,
+            tmin=tmin,
+            tmax=tmax,
+        )
+
+    def is_valid(self, dataset):
+        ret = True
+        if not (dataset.paradigm == "cvep"):
+            ret = False
+
+        # check if dataset has required events
+        if self.events:
+            if not set(self.events) <= set(dataset.event_id.keys()):
+                ret = False
+
+        return ret
+
+    @property
+    def datasets(self):
+        if self.tmax is None:
+            interval = None
+        else:
+            interval = self.tmax - self.tmin
+        return utils.dataset_search(
+            paradigm="cvep", events=self.events, interval=interval, has_all_events=True
+        )
+
+    @property
+    def scoring(self):
+        return "roc_auc"
```

Comment:
To handle the multi-level case, is it better to always use "accuracy" or to test if the number of events is greater than two and either use "roc_auc" or "accuracy"?